File: internal/content/testdata/bench/go/nc-linked-list-cycle.go
type cycleListNode struct {
	Val  int
	Next *cycleListNode
}

func has_cycle(values []int, pos int) bool {
	if len(values) == 0 {
		return false
	}
	nodes := make([]*cycleListNode, len(values))
	for i, v := range values {
		nodes[i] = &cycleListNode{Val: v}
	}
	for i := 0; i < len(nodes)-1; i++ {
		nodes[i].Next = nodes[i+1]
	}
	if pos >= 0 {
		nodes[len(nodes)-1].Next = nodes[pos]
	}
	slow, fast := nodes[0], nodes[0]
	for fast != nil && fast.Next != nil {
		slow = slow.Next
		fast = fast.Next.Next
		if slow == fast {
			return true
		}
	}
	return false
}

File: internal/content/testdata/bench/javascript/nc-linked-list-cycle.js
function has_cycle(values, pos) {
    if (!values || values.length === 0) return false;
    // Build linked list internally
    const nodes = values.map(v => ({ val: v, next: null }));
    for (let i = 0; i < nodes.length - 1; i++) {
        nodes[i].next = nodes[i + 1];
    }
    if (pos >= 0) {
        nodes[nodes.length - 1].next = nodes[pos];
    }
    // Floyd's cycle detection
    let slow = nodes[0], fast = nodes[0];
    while (fast !== null && fast.next !== null) {
        slow = slow.next;
        fast = fast.next.next;
        if (slow === fast) return true;
    }
    return false;
}
